MySQL2MySQL 主大批量写入后，再进行DDL操作，从库会正常同步数据，但是同步完后，程序中断

mysql版本：5.7.27
1.主库使用sysbench压测两张表
sysbench --db-driver=mysql --time=120 --threads=4 --report-interval=10 --mysql-host=localhost --mysql-port=3306 --mysql-socket=/data/mysql/3306/mysql.sock  --mysql-user=root --mysql-password=123456 --mysql-db=test --tables=2 --table-size=1000000 oltp_read_write prepare  
sysbench 1.0.17 (using bundled LuaJIT 2.1.0-beta2)

Initializing worker threads...

Creating table 'sbtest1'...
Creating table 'sbtest2'...
Inserting 1000000 records into 'sbtest1'
Inserting 1000000 records into 'sbtest2'
Creating a secondary index on 'sbtest1'...
Creating a secondary index on 'sbtest2'...

数据加完后，给其中一个表添加字段
mysql> alter table sbtest1 add aa int;
Query OK, 0 rows affected (10.36 sec)
Records: 0  Duplicates: 0  Warnings: 0

2.从 查询

mysql> select count(*) from sbtest1;
+----------+
| count(*) |
+----------+
|  1000000 |
+----------+
1 row in set (0.19 sec)

mysql> show create table sbtest1;
+---------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table   | Create Table                                                                                                                                                                                                                                                                                                                                                                      |
+---------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| sbtest1 | CREATE TABLE `sbtest1` (
  `id` int(11) NOT NULL AUTO_INCREMENT,
  `k` int(11) NOT NULL DEFAULT '0',
  `c` char(120) COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',
  `pad` char(60) COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',
  `aa` int(11) DEFAULT NULL,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB AUTO_INCREMENT=1000001 DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci |
+---------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.01 sec)
数据同步正常。
3.程序报错退出
{"level":"info","msg":"QueryEvent: database: test, sql: CREATE INDEX k_1 ON sbtest1(k)","time":"2020-05-07T09:03:21+08:00"}
{"level":"info","msg":"[output-mysql] ignore unsupported ddl: CREATE INDEX k_1 ON sbtest1(k)","time":"2020-05-07T09:03:21+08:00"}
{"level":"info","msg":"[binlogTailer] ddl done with gtid: 8d0d586d-676b-11ea-9628-525e60a29f1a:1-34550,e734b2fe-994d-11e9-b57a-2270be800c53:1-5,14eb9625-2164-11ea-8ac9-525e60a29f1a:1-4, stmt: CREATE INDEX k_1 ON sbtest1(k)","time":"2020-05-07T09:03:21+08:00"}
{"level":"info","msg":"QueryEvent: database: test, sql: CREATE INDEX k_2 ON sbtest2(k)","time":"2020-05-07T09:03:21+08:00"}
{"level":"info","msg":"[output-mysql] ignore unsupported ddl: CREATE INDEX k_2 ON sbtest2(k)","time":"2020-05-07T09:03:21+08:00"}
{"level":"info","msg":"[binlogTailer] ddl done with gtid: e734b2fe-994d-11e9-b57a-2270be800c53:1-5,14eb9625-2164-11ea-8ac9-525e60a29f1a:1-4,8d0d586d-676b-11ea-9628-525e60a29f1a:1-34551, stmt: CREATE INDEX k_2 ON sbtest2(k)","time":"2020-05-07T09:03:21+08:00"}
{"level":"info","msg":"QueryEvent: database: test, sql: alter table sbtest1 add aa int","time":"2020-05-07T09:03:21+08:00"}
[mysql] 2020/05/07 09:03:26 packets.go:36: read tcp 10.91.138.44:55194->10.91.138.42:3321: i/o timeout
panic: interface conversion: error is *errors.errorString, not *mysql.MySQLError

goroutine 65 [running]:
github.com/moiot/gravity/pkg/outputs/mysql.(*MySQLOutput).Execute(0xc0004f20b0, 0xc029ba2418, 0x1, 0x1, 0x2030300, 0x24fbd00)
        /usr/local/go/project/src/github.com/moiot/gravity/pkg/outputs/mysql/mysql.go:330 +0x35f4
github.com/moiot/gravity/pkg/schedulers/batch_table_scheduler.(*batchScheduler).Start.func1.1(0xc0002f4cc0, 0xc0002f4cc0)
        /usr/local/go/project/src/github.com/moiot/gravity/pkg/schedulers/batch_table_scheduler/batch_table_scheduler.go:231 +0x55
github.com/moiot/gravity/pkg/utils/retry.Do(0xc000efbfa0, 0x3, 0x3b9aca00, 0x1, 0x7f284136aeb0)
        /usr/local/go/project/src/github.com/moiot/gravity/pkg/utils/retry/retry.go:40 +0x41
github.com/moiot/gravity/pkg/schedulers/batch_table_scheduler.(*batchScheduler).Start.func1(0xc00045c280, 0xc0003c5f80, 0x2)
        /usr/local/go/project/src/github.com/moiot/gravity/pkg/schedulers/batch_table_scheduler/batch_table_scheduler.go:230 +0x6b2
created by github.com/moiot/gravity/pkg/schedulers/batch_table_scheduler.(*batchScheduler).Start
        /usr/local/go/project/src/github.com/moiot/gravity/pkg/schedulers/batch_table_scheduler/batch_table_scheduler.go:200 +0x140

[1]+  Exit 2                  ./bin/gravity --config /etc/mysql2mysql.toml >> result.log

--- a/pkg/outputs/mysql/mysql.go
+++ b/pkg/outputs/mysql/mysql.go
@@ -327,7 +327,7 @@ func (output *MySQLOutput) Execute(msgs []*core.Msg) error {
 				for _, stmt := range targetDDLs {
 					err := output.executeDDL(targetSchema, stmt)
 					if err != nil {
-						if e := errors.Cause(err).(*mysqldriver.MySQLError); e.Number == 1060 || e.Number == 1061 {
+						if e, ok := errors.Cause(err).(*mysqldriver.MySQLError); ok && (e.Number == 1060 || e.Number == 1061) {
 							log.Errorf("[output-mysql] ignore duplicate column or index. ddl: %s. err: %s", stmt, e)
 						} else {
 							log.Fatal("[output-mysql] error exec ddl: ", stmt, ". err:", err)

